scripts/password_filter.py

```diff
@@ -1,16 +1,59 @@
 import re
-def password_filter(src, dest, regex):
-	pattern = re.compile(regex)
-	pw_list = []
+import sys
+
+all_digits_regex = re.compile('^\d+$')
+def is_all_digits(password):
+    return all_digits_regex.match(password)
+
+all_alpha_regex = re.compile('^[a-zA-Z]+$')
+def is_all_alpha(password):
+    return all_alpha_regex.match(password)
+
+def password_filter(src, dest):
+        pw_set = set([])
+
 	with open(src, 'r') as f:
 		for line in f:
-			m = re.match(pattern, line)
-			if(m):
-				print m.group(1)
-				pw_list.append(m.group(1))
-	print len(pw_list)
+                # Several checks are done before the bloom filter is called:
+                #   1. password < 8 chars
+                #   2. all letters & password < 12 chars
+                #   3. all numbers & password < 12 chars
+                #
+                # If a member of any of these sets, it does not need to be
+                # a part of the filter.
+                #
+                # So, to be part of the set:
+                #   1. if password >= 8 chars.
+                #   2. if password <= 11 chars.
+                #   3. password is not all letters.
+                #   4. password is not all numbers.
+
+                # normalize to lowercase to reduce the size of the output set
+                    password = line.strip()#.lower()
```

Comment:
I was experimenting with `.lower()` to reduce the resulting dictionary even further. It's a micro-optimization that isn't needed at the current point. I'll take that out.